### PR TITLE
V3—Adornments now have models

### PR DESCRIPTION
**Pull request by bfinzer**
* Movable line and movable value adornments have models that they update during drag
* Fixes coming about from previous graph.scss factorization

### CODE REVIEW

File: v3/src/components/graph/adornments/adornment-models.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+  Adornment models are strictly MST. The keep track of the user modifications of the defaults.
```
Comment (kswenson) [inline v3/src/components/graph/adornments/adornment-models.ts:2]:
```suggestion
  Adornment models are strictly MST. They keep track of the user modifications of the defaults.
```

File: v3/src/components/graph/adornments/adornment-models.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+export interface IMovableValueModel extends Instance<typeof MovableValueModel> {}
+
+export const MovableLineModel = Adornment
+  .named('MovableLineModel')
+  .props({
+    type: 'line',
+    intercept: types.number,
+    slope: types.number,
+    pivot1: types.optional( PointModel, kInfinitePoint),
+    pivot2: types.optional( PointModel, kInfinitePoint)
```
Comment (kswenson) [inline v3/src/components/graph/adornments/adornment-models.ts:57]: Out of curiosity, do these need to be serialized or can they be determine at run time from other information?

File: v3/src/components/graph/adornments/movable-line.tsx [hunk 3]
```diff
@@ -1,100 +1,105 @@
 import React, {useCallback, useEffect, useRef, useState} from "react"
 import {drag, ScaleLinear, select} from "d3"
-import {CPLine, equationString, lineToAxisIntercepts} from "../utilities/graph_utils"
+import {equationString, lineToAxisIntercepts} from "../utilities/graph_utils"
 import "./movable-line.scss"
+import {IMovableLineModel} from "./adornment-models"
+import {autorun} from "mobx"
```
Comment (kswenson) [inline v3/src/components/graph/adornments/movable-line.tsx:6]: Conventional import ordering would have the `mobx` import in the top section with the other third-party modules and the `.scss` would be the last import.

File: v3/src/components/graph/adornments/movable-line.tsx [hunk 4]
```diff
@@ -1,100 +1,105 @@
...
-    }, [setLine, slope, x, y]),
+      model.setLine({slope: model.slope, intercept: tWorldY - model.slope * tWorldX})
+    }, [model, x, y]),
 
     continueRotation1 = useCallback((event: { x: number, y: number, dx: number, dy: number }) => {
       if (event.dx !== 0 || event.dy !== 0) {
         let isVertical = false
         const newPivot1 = {x: x.invert(event.x - 60), y: y.invert(event.y)},
-          pivot2 = props.line.pivot2 || pointsOnAxes.pt2
+          pivot2 = isFinite(model.pivot2.x) ? model.pivot2 : pointsOnAxes.current.pt2
```
Comment (kswenson) [inline v3/src/components/graph/adornments/movable-line.tsx:102]: You could also add an `isValid()` view on the `PointModel` and use `model.pivot2.isValid`.

File: v3/src/components/graph/hooks/axis-model.test.ts [hunk 5]
```diff
@@ -1,7 +0,0 @@
-import { AxisModel } from "./axis-model"
-
-describe("AxisModel", () => {
-  it("should error if AxisModel is instantiated directly", () => {
-    expect(() => AxisModel.create({ place: "left" })).toThrow()
-  })
-})
```
Comment (kswenson) [inline v3/src/components/graph/hooks/axis-model.test.ts:7]: This is still a valid test, so I think it should not be deleted.

**Review by kswenson — APPROVED** [review]
👍 Just a handful of minor comments, none of which need hold up the PR with the possible exception of undeleting the `axis-model.test.ts` file, which is still valid and serves as a useful placeholder for future tests.

File: v3/src/components/graph/adornments/adornment-models.ts (hunk 2 above)
Comment (bfinzer) [reply v3/src/components/graph/adornments/adornment-models.ts:57]: They do need to be serialized.

**Comment by bfinzer** [comment]
I've resolved your comments. I'm not going to do the merge because of conflicts.

**Pull request merged**